### ISSUE TITLE
Engine request - GZDoom

### ISSUE THREAD

**Issue by neuromancer** [dreamer/luxtorpeda#10]
Games to support:

Ultimate Doom: https://store.steampowered.com/app/2280/
Doom 2: https://store.steampowered.com/app/2300/
Master Levels of Doom 2: https://store.steampowered.com/app/9160
Heretic: https://store.steampowered.com/app/2390/
Hexen: https://store.steampowered.com/app/2360/
Hexen: Deathkings of the Dark Citadel: https://store.steampowered.com/app/2370/

Bonus:
The_Forestale: https://store.steampowered.com/app/817220/The_Forestale/

**Comment by dreamer** [comment]
It was already on the agenda on [wiki page](https://github.com/dreamer/luxtorpeda/wiki/Game-engines) - thank you for listing the app ids, it was on a TODO list :)

**Comment by dreamer** [comment]
Forestale is going to be released on Linux soon™ https://steamcommunity.com/games/817220/announcements/detail/2568649935957981841 and it seems like it'll be incompatible with upstream GZDoom going forward.

I would rather not undermine dev efforts of putting this game natively on Linux, but let's watch how the situation unfolds - maybe we can help with the Linux release somehow.

**Comment by dreamer** [comment]
Some progress in packaging gzdoom; the initial version of the package was prepared, not all listed games are supported yet: 806b6e711e16dd6791b9e59f5553236af163aca1

There are some issues, though:

- I packaged version 3.7.2,  which is the latest version without Vulkan renderer. The current version is 4.1.3, but Vulkan renderer introduced in 4.0.0 requires C++17 (perhaps unintentionally). The newest compiler available for Steam Runtime at the moment is GCC 5.4, which does not support the feature from C++17 needed by gzdoom's Vulkan renderer - to be discussed with gzdoom devs.
- gzdoom users warned me that 3.7.2 has security issues, so I am not advertising these packages in README.md yet - these are WIP, use at your own risk.
- Master Levels for Doom 2: https://store.steampowered.com/app/9160 does not work with gzdoom (not the gzdoom problem - this game is basically a DOS-based launcher for Doom maps)
- HeXen: Deathkings of the Dark Citadel (2370) comes with broken .WAD file - this can't be fixed the same way I fix it in Boxtron, but maybe the alternative solution could be found. Nevertheless - gzdoom is not capable of running this game - it seems like a regression in gzdoom.
- One more supported game Strife is https://steamdb.info/app/317040/, but I don't own it.

**Comment by dreamer** [comment]
I patched the sources to disable building of Vulkan renderer for now. Next step will be to change the GZDoom buildsystem, so I can enable Vulkan renderer again (it should be possible, as I saw Vulkan renderer providing fallback for missing C++17 feature on OpenBSD) and upstream the change, so we can build vanilla GZDoom releases, not our patched version.

Once patch will be upstreamed I will close this issue and split it into several smaller ones - one per missing game.
